drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1097 @@
+	if (retval < 0) {
+		LOG_ERR("could not read MSB of manufacturer id %d", retval);
+		return -EIO;
+	}
+
+	if (regdata != TMAG5273_MANUFACTURER_ID_MSB) {
+		LOG_ERR("unexpected manufacturer id");
+		return -EINVAL;
+	}
+
+	/* check device id*/
+	retval = i2c_reg_read_byte_dt(&drv_cfg->i2c, TMAG5273_REG_DEVICE_ID, &regdata);
+	if (retval < 0) {
+		LOG_ERR("could not read device id %d", retval);
+		return -EIO;
+	}
+
+	tmag5273_read_device_status(dev);
+
+	retval = tmag5273_reset_device_status(dev);
+	if (retval < 0) {
+		LOG_ERR("could not reset DEVICE_STATUS register %d", retval);
+		return -EIO;
+	}
+
+	drv_data->version = FIELD_GET(TMAG5273_VER_MSK, regdata);
+
+	/* magnetic measurement range based on version, apply correct one */
+	switch (drv_data->version) {
+	case TMAG5273_VER_TMAG5273X1:
+		drv_data->xyz_range = (drv_cfg->meas_range != TMAG5273_DT_AXIS_RANGE_LOW)
+					      ? TMAG5273_MEAS_RANGE_HIGH_MT_VER1
+					      : TMAG5273_MEAS_RANGE_LOW_MT_VER1;
+		break;
+	case TMAG5273_VER_TMAG5273X2:
+		drv_data->xyz_range = (drv_cfg->meas_range != TMAG5273_DT_AXIS_RANGE_LOW)
+					      ? TMAG5273_MEAS_RANGE_HIGH_MT_VER2
+					      : TMAG5273_MEAS_RANGE_LOW_MT_VER2;
+		break;
+	default:
+		LOG_ERR("unexpected version");
+		return -EINVAL;
+	}
+
+	/* deactivate interrupts if necessary (will be overwritten if needed) */
+	regdata = FIELD_PREP(TMAG5273_INT_MODE_MSK, TMAG5273_INT_MODE_NONE) |
+		  FIELD_PREP(TMAG5273_INT_MASK_INTB_MSK,
+			     (drv_cfg->use_int_gpio ? TMAG5273_INT_MASK_INTB_PIN_ENABLED
+						    : TMAG5273_INT_MASK_INTB_PIN_MASKED));
+
+	retval = i2c_reg_write_byte_dt(&drv_cfg->i2c, TMAG5273_REG_INT_CONFIG_1, regdata);
+	if (retval < 0) {
+		LOG_ERR("error deactivating interrupts");
+		return -EIO;
+	}
+
+	/* set settings */
+	retval = tmag5273_init_sensor_settings(drv_cfg);
+	if (retval < 0) {
+		LOG_ERR("error setting sensor configuration %d", retval);
+		return retval;
+	}
+
+	retval = tmag5273_init_device_config(dev);
+	if (retval < 0) {
+		LOG_ERR("error setting device configuration %d", retval);
+		return retval;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api tmag5273_driver_api = {
+	.attr_set = tmag5273_attr_set,
+	.attr_get = tmag5273_attr_get,
+	.sample_fetch = tmag5273_sample_fetch,
+	.channel_get = tmag5273_channel_get,
+};
+
+/** Instantiation macro */
+#define TMAG5273_DEFINE(inst)                                                                      \
+	BUILD_ASSERT(IS_ENABLED(CONFIG_CRC) ||                                                     \
+			     (!IS_ENABLED(CONFIG_CRC) && (DT_INST_PROP(inst, crc_enabled) == 0)),  \
+		     "CRC support necessary");                                                     \
+	BUILD_ASSERT(!DT_INST_PROP(inst, trigger_conversion_via_int) ||                            \
+			     DT_INST_NODE_HAS_PROP(inst, int_gpios),                               \
+		     "int-gpios need to be defined");                                              \
```

Comment:
maybe `trigger-conversion-via-int requires int-gpios to be defined`

Reply:
But it's checked here? If `trigger-conversion-via-int` is not set the assert does not fail, and otherwise the existence of `int-gpios` is checked.

Reply:
yeah I mean I'm suggesting to change the error message to mention the full condition

Reply:
trying to avoid patterns like https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/logging/log_core.c#L67-L70 (which somehow I bump into all the time) where you get a build error with unclear trigger... actually let me send a PR for that one, not sure why always suffered through it